ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/ProjectOperator.java

```diff
@@ -163,17 +86,17 @@ public Object next() {
     if (row == null) {
       return null;
     }
-    if (isSelectStar) {
+    if (logicalNode.getIsSelectStar()) {
       return createRow(row);
     }
-    final GenericRow intermediate = preSelectTransform.apply(row);
+    final GenericRow intermediate = getPreSelectTransform().apply(row);
```

Comment:
If you just apply the transform immediately, can you just call it directly rather than returning a function?